src/ebmc/show_formula_solver.h

```diff
@@ -0,0 +1,70 @@
+/*******************************************************************\
+
+Module: Show Formula Solver
+
+Author: Daniel Kroening, kroening@kroening.com
+
+\*******************************************************************/
+
+/// \file
+/// Show Formula Solver
+
+#ifndef EBMC_SHOW_FORMULA_SOLVER_H
+#define EBMC_SHOW_FORMULA_SOLVER_H
+
+#include <solvers/stack_decision_procedure.h>
+
+class show_formula_solvert : public stack_decision_proceduret
+{
+public:
+  // console
+  show_formula_solvert();
+
+  // file
+  explicit show_formula_solvert(std::ostream &_out) : out(_out), console(false)
+  {
+  }
+
+  void set_to(const exprt &, bool value) override;
+  exprt handle(const exprt &) override;
+  exprt get(const exprt &) const override;
+  void print_assignment(std::ostream &) const override
+  {
+  }
+
+  std::string decision_procedure_text() const override
+  {
+    return "show-formula solver";
+  }
+
+  std::size_t get_number_of_solver_calls() const override
+  {
+    return 0;
+  }
+
+  ~show_formula_solvert()
+  {
+  }
+
+  void push(const std::vector<exprt> &) override;
```

Comment:
Nit pick: why is this one placed in the implementation file when the below ones are inline?

Reply:
I am considering to output the `push` calls, which would then require exposing the include for `console.h`.

Reply:
But there's the other variant of `push` that's not in the .cpp file? Just asking for as little surprise as possible.

Reply:
Variant moved.